app/src/main/java/teamagam/ConnectionTest/ValidatorAsyncTask.java

```diff
@@ -0,0 +1,40 @@
+package teamagam.ConnectionTest;
+
+import android.os.AsyncTask;
+
+public class ValidatorAsyncTask extends AsyncTask<String, Void, Boolean> {
+
+    private Validator mValidator;
+    private Displayer mDisplayer;
+
+    public ValidatorAsyncTask(Validator validator, Displayer displayer) {
+        mValidator = validator;
+        mDisplayer = displayer;
+    }
+
+    @Override
+    protected Boolean doInBackground(String... params) {
+        if (mValidator.validate(params[0])) {
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    protected void onPostExecute(Boolean validatorResult) {
+        if (validatorResult) {
+            mDisplayer.displayOK();
+        } else {
+            mDisplayer.displayFaild();
+        }
+    }
+
+
+    public interface Displayer {
+        void displayOK();
+
+        void displayFaild();
+    }
+
+
+}
```

Comment:
PERFECT!!! Beautiful isn't it? You like it or not? I think it's the most elegant class in the world. This is how code should look like!